Tests/SonarScanner.MSBuild.TFS.Test/Classic/BinaryToXmlCoverageReportConverterTests.cs

```diff
@@ -309,6 +311,37 @@ public void Initialize_CanGetGetExeToolPathFromSetupConfigurationForBuildAgent()
         public void GetRegistryPath_When32BitProcess_Returns32BitPath() =>
             BinaryToXmlCoverageReportConverter.GetVsRegistryPath(false).Should().Be(@"SOFTWARE\Microsoft\VisualStudio");
 
+        [CodeCoverageExeTestMethod]
+        [DeploymentItem(@"Resources\Sample.coverage")]
+        [DeploymentItem(@"Resources\Expected.xmlcoverage")]
+        public void Conv_ConvertToXml_ToolConvertsSampleFile()
+        {
+            // Arrange
+            var logger = new TestLogger();
+            var config = new AnalysisConfig();
+            config.SetVsCoverageConverterToolPath(CodeCoverageExeTestMethodAttribute.FindCodeCoverageExe());
+            var reporter = new BinaryToXmlCoverageReportConverter(logger, config);
+            reporter.Initialize();
+            var inputFilePath = $"{Environment.CurrentDirectory}\\Sample.coverage";
+            var outputFilePath = $"{Environment.CurrentDirectory}\\Sample.xmlcoverage";
+            var expectedOutputFilePath = $"{Environment.CurrentDirectory}\\Expected.xmlcoverage";
+            File.Exists(inputFilePath).Should().BeTrue();
+            File.Exists(outputFilePath).Should().BeFalse();
+            File.Exists(expectedOutputFilePath).Should().BeTrue();
+
+            // Act
+            var actual = reporter.ConvertToXml(inputFilePath, outputFilePath);
+
+            // Assert
+            actual.Should().BeTrue();
+            File.Exists(outputFilePath).Should().BeTrue();
+            var actualContent = XDocument.Load(outputFilePath);
+            var expectedContent = XDocument.Load(expectedOutputFilePath);
+            // All tags and attributes must appear in the same order for actual and expected. Comments, whitespace, and the like is ignored in the assertion.
+            actualContent.DescendantNodes().OfType<XElement>().Should().SatisfyRespectively(
+                expectedContent.DescendantNodes().OfType<XElement>().Select<XElement, Action<XElement>>(expected => actual => actual.Should().Be(expected)));
```

Comment:
I tested the assertion by hand:
* I removed, renamed and change the position of tags -> all assertion errors
* I did the same for attributes -> all assertion errors